build.sh

```diff
@@ -109,6 +114,38 @@ function parse_command()
     done
 }
 
+function detect_arch_and_os() {
+    if [ -z "$target_arch" ]; then
+        target_arch=$(uname -m)
+    fi
+    
+    raw_arch="$target_arch"
+
+    case "$raw_arch" in
+        x86_64)          product_arch="amd64" ;;
+        aarch64)         product_arch="arm64" ;;
+        armv7l|armv7hl)  product_arch="armv7" ;;
+        riscv64)         product_arch="riscv64" ;;
+        ppc64le)         product_arch="ppc64le" ;;
+        s390x)           product_arch="s390x" ;;
+        i686|i386)       product_arch="386" ;;
+        *)               product_arch="$raw_arch" ;;
+    esac
```

Comment:
This should inspect the upstream packages instead of hardcoding certain architectures. The default of passing arch as-is to `rpmbuild` may lead to unexpected errors, and it should be handled a bit more gracefully.